lib/fake_ifdh.py

```diff
@@ -62,7 +62,7 @@ def checkToken(tokenfile: str) -> bool:
     cmd = f"decode_token.sh -e exp {tokenfile} 2>/dev/null"
     with os.popen(cmd, "r") as f:
         exp_time = f.read()
-    return exp_time and ((int(exp_time) - time.time()) > 60)
+    return bool(exp_time) and ((int(exp_time) - time.time()) > 60)
```

Comment:
`bool(exp_time)` isn't really the right answer here, although "should" mostly work. Actually I'm not sure what the point of that guard is, all it will do is make it even harder to debug issues since it will swallow any errors from running decode_token, which we don't otherwise check for (but should). Instead we'd get a message about a problem running `htgettoken` which would be less than helpful. But I digress.

Reply:
Okay - I think I resolved this in the next commit.  Good catch - this is a big issue as written (and as I modified it).  For example:

```python
def func():
    return "foo" and (int('') > 3)

print(func())
``` 
raises the correct `ValueError`, but the following:

```python
def func():
    return '' and (int('') > 3)

print(func())
```
gives an empty string.  So this is a bug we've been masking this whole time.  I've updated the code and the unit test.

Reply:
Yes, exactly. Worst case for most errors here we'd just end up getting a new token anyways, which I suppose was the initial thinking, but it's those couple other error conditions where everything goes haywire that we need to catch.

Your fixes look good, I like the additional error text.